snapcraft/models/project.py

```diff
@@ -171,6 +172,32 @@ def _validate_architectures_all_keyword(architectures):
             )
 
 
+def root_packages_transform(yaml_data: dict[str, Any]) -> dict[str, Any]:
```

Comment:
```suggestion
def transform_root_packages(yaml_data: dict[str, Any]) -> dict[str, Any]:
```
Or if you prefer, `apply_root_packages_transform`. Anything that's "<verb>_<noun> and descriptive 🙂

Reply:
I would use apply, so it keeps the same semantics as `apply_extensions`